core/client/app/templates/application.hbs

```diff
@@ -2,11 +2,11 @@
 
 {{gh-alerts notify="topNotificationChange"}}
 
-<div class="gh-viewport">
+<div class="gh-viewport {{autoNavClass}}">
 
-    {{#unless signedOut}}
-        {{partial "nav-menu"}}
-    {{/unless}}
+    {{#if session.isAuthenticated}}
```

Comment:
This will cause the nav menu to be rendered too soon and it will awkwardly appear on the signin (etc.) screen while the route transition processes.  It will also be removed too soon on log out for the same reason.